plugins/catalog-backend/src/ingestion/LocationReaders.ts

```diff
@@ -248,6 +248,29 @@ export class LocationReaders implements LocationReader {
       return undefined;
     }
 
+    let handled = false;
+    for (const processor of processors) {
+      if (processor.validateEntityKind) {
+        try {
+          handled = await processor.validateEntityKind(current);
+          if (handled) {
+            break;
+          }
+        } catch (e) {
+          const message = `Processor ${processor.constructor.name} threw an error while validating the entity ${kind}:${namespace}/${name} at ${item.location.type} ${item.location.target}, ${e}`;
+          emit(result.inputError(item.location, message));
```

Comment:
obv this whole repeated junk needs to be done better, but it is what it is right now